ee/session_recordings/session_summary/llm/consume.py

```diff
@@ -0,0 +1,240 @@
+                continue
+            accumulated_content += raw_content
+            try:
+                intermediate_summary = _convert_llm_content_to_session_summary_json(
+                    content=accumulated_content,
+                    allowed_event_ids=allowed_event_ids,
+                    session_id=session_id,
+                    simplified_events_mapping=simplified_events_mapping,
+                    simplified_events_columns=simplified_events_columns,
+                    url_mapping_reversed=url_mapping_reversed,
+                    window_mapping_reversed=window_mapping_reversed,
+                    session_metadata=session_metadata,
+                    summary_prompt=summary_prompt,
+                )
+                if not intermediate_summary:
+                    continue
+                # If parsing succeeds, yield the new chunk
+                sse_event_to_send = _serialize_to_sse_event(
+                    event_label="session-summary-stream", event_data=intermediate_summary
+                )
+                yield sse_event_to_send
+            except SummaryValidationError:
+                # We can except incorrect schemas because of incomplete chunks, ok to skip some.
+                # The stream should be retried only at the very end, when we have all the data.
+                continue
+            except ValueError as err:
+                # The only way to raise ValueError is data hallucinations and inconsistencies (like missing mapping data).
+                # Such exceptions should be retried as early as possible to decrease the latency of the stream.
+                logger.exception(
+                    f"Hallucinated data or inconsistencies in the session summary for session_id {session_id}: {err}"
+                )
+                raise ExceptionToRetry()
+    except (openai.APIError, openai.APITimeoutError, openai.RateLimitError) as err:
+        logger.exception(f"Error streaming LLM for session_id {session_id} by user {user.pk}: {err}")
+        raise ExceptionToRetry()
+    # Final validation of accumulated content (to decide if to retry the whole stream or not)
+    try:
+        if accumulated_usage:
+            TOKENS_IN_PROMPT_HISTOGRAM.observe(accumulated_usage)
+        final_summary = _convert_llm_content_to_session_summary_json(
+            content=accumulated_content,
+            allowed_event_ids=allowed_event_ids,
+            session_id=session_id,
+            simplified_events_mapping=simplified_events_mapping,
+            simplified_events_columns=simplified_events_columns,
+            url_mapping_reversed=url_mapping_reversed,
+            window_mapping_reversed=window_mapping_reversed,
+            session_metadata=session_metadata,
+            summary_prompt=summary_prompt,
+            final_validation=True,
+        )
+        if not final_summary:
+            logger.exception(f"Final LLM content validation failed for session_id {session_id}")
```

Comment:
and could capture these to `posthoganalytics.capture_exception`

i've been adding `replay_feature` property to them all so I can find them more easily

Reply:
TIL, added a TODO, will implement together with timing/tokensize tracking.